MeasurementProtocol.php

```diff
@@ -47,6 +50,24 @@ public function request()
             $request->setAnonymizeIp(1);
         }
 
+        if ($this->autoSetClientId) {
+            $clientId = $this->extractClientIdFromCookie();
+            if (!empty($clientId)) {
+                $request->setClientId($clientId);
+            }
+        }
+
         return $request;
     }
+
+    /**
+     * @return string
+     */
+    protected function extractClientIdFromCookie()
```

Comment:
Please, pay attention to the protected method. Maybe you prefer to have a private methods, so class could be extended via decorators?

Reply:
No, that’s fine.